desc/compute/_core.py

```diff
@@ -317,12 +317,12 @@ def compute_rotational_transform(
     if data is None:
         data = {}
 
+    grid = R_transform.grid
     if iota is not None:
-        data["iota"] = iota.compute(i_l, dr=0)
-        data["iota_r"] = iota.compute(i_l, dr=1)
+        data["iota"] = iota.compute(i_l, grid, dr=0)
```

Comment:
I don't think this is necessary, and may actually cause issues with recompiling on each iteration. ``iota.compute`` should use the grid assigned to the profile object, regardless of type.

Reply:
Should I remove the grid in current.compute from the same function? I had added that to do an interactive spline solve with it.

Reply:
was it giving errors without passing in the grid? If so it might be an issue with the profile classes

Reply:
I don't think there's anything wrong with spline. Passing in the grid gives you more flexibility. Which was convenient for me because I could generate a spline that has knots at a bunch of rho values I wanted, and call `compute` on it with quadrature/concentric grid later and it would still work.